### PR TITLE
Bundle noise weights and mask into `WeightOperator`

### PR DESCRIPTION
**Pull request by sbiquard**

### CODE REVIEW

File: src/furax/mapmaking/_model.py [hunk 1]
```diff
@@ -18,6 +18,21 @@
...
+    weight: AbstractLinearOperator  # symmetric
+    mask: MaskOperator
+
+    @classmethod
+    def create(cls, weight: AbstractLinearOperator, mask: MaskOperator) -> Self:
+        return cls(weight, mask, in_structure=mask.in_structure)
+
+    def mv(self, x: PyTree[Inexact[jax.Array, '...']]) -> PyTree[Inexact[jax.Array, '...']]:
+        return self.mask(self.weight(self.mask(x)))
+
```
Comment (Wuhyun) [inline src/furax/mapmaking/_model.py:20]: Very small optimisation (likely not very crucial) that could be done here would be:
When W is diagonal, `Z W Z = W Z`, so we can do `self.weight(self.mask(x))`

File: src/furax/mapmaking/_model.py [hunk 2]
```diff
@@ -18,6 +18,21 @@
 from .templates import ATOPProjectionOperator
 
 
+@symmetric
+class WeightOperator(AbstractLinearOperator):
+    """Masked noise weight `Z W Z` as a single operator."""
```
Comment (Wuhyun) [inline src/furax/mapmaking/_model.py]: Is the name `Z` consistent with our notation elsewhere or in the literature somewhere?
If it's new, we could explain it a little bit here!

**Review by Wuhyun — DISMISSED** [review]
Looks good to me, just a few minor comments, but feel free to go ahead! Thanks

File: src/furax/mapmaking/_model.py (hunk 1 above)
Comment (sbiquard) [reply src/furax/mapmaking/_model.py:20]: done!

File: src/furax/mapmaking/_model.py (hunk 2 above)
Comment (sbiquard) [reply src/furax/mapmaking/_model.py]: changed to `M` in the end
